src/main/java/io/xpring/xrpl/model/XRPMemo.java

```diff
@@ -0,0 +1,61 @@
+package io.xpring.xrpl.model;
+
+import com.google.protobuf.ByteString;
+import org.xrpl.rpc.v1.Memo;
+import org.immutables.value.Value;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a memo on the XRPLedger.
+ * @see "https://xrpl.org/transaction-common-fields.html#memos-field"
+ */
+@Value.Immutable
+public interface XRPMemo {
+    static ImmutableXRPMemo.Builder builder() {
+        return ImmutableXRPMemo.builder();
+    }
+
+    /**
+     * @return Arbitrary hex value, conventionally containing the content of the memo.
+     */
+    @Nullable  byte[] data();
+
+    /**
+     * @return Hex value representing characters allowed in URLs.
+     * Conventionally containing information on how the memo is encoded, for example as a MIME type.
+     */
+    @Nullable byte[] format();
+
+    /**
+     * @return Hex value representing characters allowed in URLs.
+     * Conventionally, a unique relation (according to RFC 5988) that defines the format of this memo.
+     */
+    @Nullable byte[] type();
+
+    static XRPMemo from(Memo memo) {
+        byte[] data;
```

Comment:
I'd build these up one at a time to make it easier to read. Also, the null value is implicitly assigned, though you can make it explicit. 

```
byte[] data; // Or if you want to be super clear: byte[] data= null; 
if (memo.getMemoData().getValue() != ByteString.EMPTY) {
  data = memo.getMemoData().getValue().toByteArray();
}
        // Intentional blank line; let reader know we're done with mucking around with `data`
byte [] format; 
// and so on
```

If you really wanted to be succinct there, you could just use a ternary operator and do the assignment in two statements:
```
ByteString memoData = memo.getMemoData().getValue();
byte[] data = memoData == ByteString.EMPTY ? null : memoData.toByteArray()
        // Intentional blank line; let reader know we're done with mucking around with `data`
ByteString format = ...
```

I tend to prefer the second though some people find ternary operators hard to read.

Reply:
wayyyyyy prettier thank you